File: src/java/org/voegtle/weatherstation/server/RepairServlet.java
package org.voegtle.weatherstation.server;

import org.voegtle.weatherstation.server.logic.WeatherDataRepair;
import org.voegtle.weatherstation.server.persistence.SmoothedWeatherDataSet;
import org.voegtle.weatherstation.server.request.RepairUrlParameter;
import org.voegtle.weatherstation.server.request.ResponseCode;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;
import java.util.List;

public class RepairServlet extends AbstractServlet {

  @Override
  protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {

    RepairUrlParameter param = new RepairUrlParameter(request, locationProperties.getDateUtil());

    if (isSecretValid(param.getSecret())) {
      WeatherDataRepair repairService = new WeatherDataRepair(pm, locationProperties);
      List<SmoothedWeatherDataSet> repaired = repairService.repair(param.getBegin(), param.getEnd());
      returnDetailedResult(response, repaired, false);
    } else {
      returnResult(response, ResponseCode.NOT_AUTHORIZED);
    }
  }

}

File: src/main/java/org/voegtle/weatherstation/server/RepairServlet.kt
package org.voegtle.weatherstation.server

import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.voegtle.weatherstation.server.logic.WeatherDataRepair
import org.voegtle.weatherstation.server.request.RepairUrlParameter
import org.voegtle.weatherstation.server.request.ResponseCode
import java.io.IOException

class RepairServlet : AbstractServlet() {

  @Throws(ServletException::class, IOException::class)
  override fun doGet(request: HttpServletRequest, response: HttpServletResponse) {

    val param = RepairUrlParameter(request, locationProperties!!.dateUtil)

    if (isSecretValid(param.secret)) {
      val repairService = WeatherDataRepair(pm, locationProperties!!)
      val repaired = repairService.repair(param.begin!!, param.end)
      returnDetailedResult(response, repaired, false)
    } else {
      returnResult(response, ResponseCode.NOT_AUTHORIZED)
    }
  }

}
